src/components/Button/Button.styles.ts

```diff
@@ -0,0 +1,127 @@
+import { Button as MuiButton, CircularProgress } from '@mui/material';
+import { styled } from '@mui/material/styles';
+
+import {
+  blueScale,
+  redScale,
+  greenScale,
+  greyScale,
+} from '../../style-library/types/theme.helpers';
+
+import { SIZE_CONFIGS, SHAPE_CONFIGS } from './Button.constants';
+
+import type { StyledButtonProps } from './Button.types';
+
+// Map variant → palette
+const VARIANT_COLORS = {
+  primary: {
+    main: blueScale[600],
+    light: blueScale[50],
+    hover: blueScale[700],
+    active: blueScale[800],
+  },
+  secondary: {
+    main: greyScale[600],
+    light: greyScale[50],
+    hover: greyScale[700],
+    active: greyScale[800],
+  },
+  success: {
+    main: greenScale[600],
+    light: greenScale[50],
+    hover: greenScale[700],
+    active: greenScale[800],
+  },
+  error: { main: redScale[600], light: redScale[50], hover: redScale[700], active: redScale[800] },
+} as const;
+
+export const StyledButton = styled(MuiButton, {
+  shouldForwardProp: (prop) => !prop.toString().startsWith('$'),
+})<StyledButtonProps>(({ theme, $variant, $size, $shape, $loading, $fullWidth }) => {
+  const colors = VARIANT_COLORS[$variant];
+  const size = SIZE_CONFIGS[$size];
+  const shape = SHAPE_CONFIGS[$shape];
+
+  return {
+    height: size.height,
+    minWidth: $fullWidth ? '100%' : size.minWidth,
+    padding: size.padding,
+    fontSize: size.fontSize,
+    fontWeight: 500,
+    textTransform: 'none',
+    borderRadius: shape.borderRadius,
```

Comment:
borderRadius is supplied from SHAPE_CONFIGS using hardcoded px strings. The file should use theme.shape or theme.spacing tokens instead of hard-coded px values.

**Sugerencia:**
```typescript
Map shape tokens to theme values here, e.g. borderRadius: theme.shape.borderRadius for rounded, or use numeric tokens and calculate: borderRadius: theme.spacing(shape.borderRadiusUnit).
```